connectivity/manifests/allow-all-except-world.yaml

```diff
@@ -19,9 +19,9 @@ spec:
   # This change prevents failing the connectivity
   # test for such environments.
   - toPorts:
-      - ports:
-          - port: "53"
-            protocol: UDP
+    - ports:
+      - port: "53"
+        protocol: ANY
```

Comment:
Why are we opening the policy rules for all protocols if the intention is to only allow DNS traffic? Did you try adding a similar rule for TCP?

Reply:
Thanks for asking. I should have mentioned it already. I was searching for the root cause. I did a test with TCP it didn't affect the result. The problem is that the policies with `FQDNS` do not work. Of course, I am talking about the environment with the `node-local-dns` It looks like it cannot detect the protocol when only the `client-egress-to-fqdns-one-one-one-one` or the `client-egress-only-dns` policies are applied.

I added the DNS rule (the part below) for the above policies. But the problem wasn't that something was missing.

```yaml
      rules:
        dns:
        - matchPattern: "*"
```

Then, I noticed that Cilium couldn't validate the protocols.

We might add a test environment with `node-local-dns` deployed. I can help with it.

I noticed this problem while I was testing the new release of Cilium.

Reply:
Worth noting that if this is set to `ANY` and there is a `rules` stanza, Cilium only supports redirecting traffic for TCP and UDP so I believe this is effectively the same as having another rule for TCP specifically.